src/btree/bt_cursor.c

```diff
@@ -499,6 +474,31 @@ __cursor_search_neighboring(WT_CURSOR_BTREE *cbt, WT_CURFILE_STATE *state, int *
     return (ret);
 }
 
+/*
+ * __wt_cursor_row_search --
+ *     Row-store search from a cursor.
+ */
+int
+__wt_cursor_row_search(WT_CURSOR_BTREE *cbt, bool insert, WT_REF *leaf, bool *leaf_foundp)
```

Comment:
The naming here makes this change to public look not quite right. Nearly all of the public (non-static) functions in this file have the prefix __wt_btcur, so adding one with the name __wt_cursor feels like an interface or layering violation.

I wonder if the new caller should actually be in btree_inline.h - does that function actually need to be inlined? It's complex enough that I doubt getting rid of a function call overhead is going to make much of a perf difference to the application.

Reply:
Actually, thinking about this more. Exposing a row-store specific function from this file does feel quite wrong to me. The btcur API should protect the calling code from a row/col distinction.

Reply:
Fixed this!I have moved the `__wt_btcur_bounds_row_position` function inside to bt_cursor.c, I agree, I don't think we need that function to be inlined.